### PR TITLE
Improve #1354 "Moderator Options" time difference

### PR DESCRIPTION
**Pull request by Sama34**
Improve #1354 "Moderator Options" time difference

### CODE REVIEW

File: admin/modules/user/users.php [hunk 1]
```diff
@@ -1526,8 +1542,24 @@ function toggleAction()
 		$mybb->input['moderateposting'] = 1;
 		if($user['moderationtime'] != 0)
 		{
-			$expired = my_date('relative', $user['moderationtime'], '', 2);
-			$existing_info = $lang->sprintf($lang->moderate_length, $expired);
+			$remaining = $user['moderationtime']-TIME_NOW;
+			$expired = nice_time($remaining, array('seconds' => false));
+
+			$color = 'inherit';
+			if($remaining <= 0 || $remaining < 3600)
```
Comment (Destroy666x) [inline admin/modules/user/users.php]: What's the point of checking `$remaining <= 0`? If it's lower than 3600 it's also lower than 0.

File: admin/modules/user/users.php [hunk 2]
```diff
@@ -1371,8 +1371,24 @@
...
+			{
+				$color = 'red';
+			}
+			elseif($remaining < 86400)
+			{
+				$color = 'maroon';
+			}
+			elseif($remaining < 604800)
+			{
+				$color = 'green';
```
Comment (Destroy666x) [inline admin/modules/user/users.php]: I think yellow would be more suitable here, green indicates something good and I suppose this should be a warning.

**Comment by Sama34** [comment]
The code was based off (not to say it was copy/pasting) from existing core in (IIRC) ''modcp.php'.

**Comment by Destroy666x** [comment]
@Sama34 , even if there are conditionals like that there is no reason to duplicate the mistakes.

As for the colors, I see, they're consistent now. We could change the green to dark yellow or orange in the 4 files which set it based on time. But I guess this can be done in another issue.

**Comment by Sama34** [comment]
Done.

**Pull request merged**